.evg.yml

```diff
@@ -3,100 +3,196 @@ functions:
-              if [ $? -ne 0 ]; then
-                echo "failure" > commit_state
-                exit 0
-              fi
-            done
-
-            echo "success" > commit_state
-      - func: "set_github_build_status"
+            set -e
+            cd stitch-ios-sdk/StitchCore
 
-  - name: run_tests
+            echo "running swift core tests"
+            
+            swift package generate-xcodeproj
+            xcodebuild test -scheme SwiftCode-Package -enableCodeCoverage YES
+            xcov --legacy_support xccoverage --source_directory Sources/ --exclude_targets ExtendedJSON.framework,JWT.framework,StitchCore_AdminClient.framework,Swifter.framework
+      - func: "publish_codecov"
+  - name: run_ios_tests
     commands:
       - func: "fetch_source"
+      - func: "setup_xcov"
+      - func: "setup_mongod"
+      - func: "setup_stitch"
       - command: shell.exec
-        silent: true
         params:
+          shell: "bash"
           script: |
             set -e
-            cd stitch-ios-sdk
-            git submodule update --init --recursive
-
-            echo "packaging source files"
-            tar -czf ../source.tgz .
-
-            urlencodepipe() {
-              local LANG=C; local c; while IFS= read -r c; do
-                case $c in [a-zA-Z0-9.~_-]) printf "$c"; continue ;; esac
-                printf "$c" | od -An -tx1 | tr ' ' % | tr -d '\n'
-              done <<EOF
-            $(fold -w1)
-            EOF
-              echo
-            }
+            cd stitch-ios-sdk/StitchCore-iOS
 
-            urlencode() { printf "$*" | urlencodepipe ;}
-
-            echo "running tests remotely"
-            TEST_DESTINATION=`urlencode "platform=iOS Simulator,name=iPhone 7 Plus"`
-            TEST_SCHEMES=`urlencode "StitchCore,ExtendedJson,MongoDBService"`
-            URL="${device_farm_url}?test_type=ios&test_destination=$TEST_DESTINATION&test_schemes=$TEST_SCHEMES"
-            RESULT=`cat ../source.tgz | curl $URL -H "Authorization: ${device_farm_secret}" --data-binary @-`
-            echo $RESULT
-            echo $RESULT | grep '!!ALL TESTS PASSED!!'
+            echo "running iOS tests"
+            xcodebuild test -workspace ../Stitch.xcworkspace/ -scheme StitchCore-iOS -configuration Debug -enableCodeCoverage YES -destination "platform=iOS Simulator,name=iPhone 8,OS=11.2"
+            xcov --legacy_support xccoverage --exclude_targets ExtendedJSON.framework,JWT.framework,StitchCore_AdminClient.framework,Swifter.framework
+      - func: "publish_codecov"
+post:
+  - command: attach.xunit_results
+    params:
+      files:
+        - "./stitch-ios-sdk/StitchCore/xcov_report/**/*"
+        - "./stitch-ios-sdk/StitchCore-iOS/xcov_report/**/*"
+  - command: s3.put
+    params:
+      aws_key: ${test_artifacts_aws_key}
+      aws_secret: ${test_artifacts_aws_secret}
+      remote_file: ${build_id}/report
```

Comment:
[major] Missing `local_files_include_filter`. See java evergreen file. You need to make sure each file has a unique base name as well. It gets concatenated to the remote_file base name.